superagi/controllers/knowledges.py

```diff
@@ -0,0 +1,165 @@
+    vector_database_index = VectordbIndices.get_vector_index_from_id(db.session, knowledge_data.vector_db_index_id)
+    vector_database = Vectordbs.get_vector_db_from_id(db.session, vector_database_index.vector_db_id)
+    knowledge = {
+        "name": knowledge_data.name,
+        "description": knowledge_data.description,
+        "vector_database_index": {
+            "id": vector_database_index.id,
+            "name": vector_database_index.name
+        },
+        "vector_database": vector_database.name,
+        "installation_type": vector_database_index.state
+    }
+    knowledge_config = KnowledgeConfigs.get_knowledge_config_from_knowledge_id(db.session, knowledge_id)
+    knowledge_data_with_config = knowledge | knowledge_config
+    return knowledge_data_with_config
+
+@router.post("/add_or_update/data")
+def add_update_user_knowledge(knowledge_data: dict, organisation = Depends(get_user_organisation)):
+    knowledge_data["organisation_id"] = organisation.id
+    knowledge_data["contributed_by"] = organisation.name
+    knowledge = Knowledges.add_update_knowledges(db.session, knowledge_data)
+    return {"success": True, "id": knowledge.id}
+
+@router.post("/delete/{knowledge_id}")
+def delete_user_knowledge(knowledge_id: int):
+    try:
+        Knowledges.delete_knowledge(db.session, knowledge_id)
+        return {"success": True}
+    except:
+        return {"success": False}
+
+@router.get("/install/{knowledge_name}/index/{vector_db_index_id}")
+def install_selected_knowledge(knowledge_name: str, vector_db_index_id: int, organisation = Depends(get_user_organisation)):
+    vector_db_index = VectordbIndices.get_vector_index_from_id(db.session, vector_db_index_id)
+    selected_knowledge = Knowledges.fetch_knowledge_details_marketplace(knowledge_name)
+    selected_knowledge_config = KnowledgeConfigs.fetch_knowledge_config_details_marketplace(selected_knowledge['id'])
+    file_chunks = S3Helper().get_json_file(selected_knowledge_config["file_path"])
+    vector = Vectordbs.get_vector_db_from_id(db.session, vector_db_index.vector_db_id)
+    db_creds = VectordbConfigs.get_vector_db_config_from_db_id(db.session, vector.id)
+    upsert_data = VectorEmbeddingFactory.build_vector_storge(vector.db_type).get_vector_embeddings_from_chunks(file_chunks)
+    try:
+        vector_db_storage = VectorFactory.build_vector_storage(vector.db_type, vector_db_index.name, **db_creds)
+        vector_db_storage.add_embeddings_to_vector_db(upsert_data)
+    except:
+        return {"success": False}
+    selected_knowledge_data = {
+        "id": -1,
+        "name": selected_knowledge["name"],
+        "description": selected_knowledge["description"],
+        "index_id": vector_db_index_id,
+        "organisation_id": organisation.id,
+        "contributed_by": selected_knowledge["contributed_by"],
+    }
+    new_knowledge = Knowledges.add_update_knowledges(db.session, selected_knowledge_data)
+    selected_knowledge_config.pop("file_path")
```

Comment:
Can you make a constant var and use it instead of hard coding it for increasing readability.